src/base/bittorrent/session.cpp

```diff
@@ -141,19 +141,55 @@ namespace
         return true;
     }
 
+#ifdef Q_OS_WIN
+    // Remove this after 4.2.5 && if at least one month has passed from v4.2.3
+    // Check the commit that introduced this function and identify all other pieces of code that
+    // need removal alongside this one.
+    void fixBrokenSavePath(QByteArray &data, CreateTorrentParams &torrentParams, lt::bdecode_node &root)
+    {
+        const QString path = fromLTString(root.dict_find_string_value("save_path"));
+        const int index = path.indexOf(QLatin1String("//"));
+        if (index < 1)
+            return;
+
+        const QString goodPath = path.mid(index);
+        torrentParams.needSaveResumeData = true;
+        lt::entry entry(root);
+        entry["save_path"] = goodPath.toStdString();
+        data.clear();
+        lt::bencode(std::back_inserter(data), entry);
+        lt::error_code ec;
+#if (LIBTORRENT_VERSION_NUM < 10200)
+        lt::bdecode(data.constData(), (data.constData() + data.size()), root, ec);
+#else
+        root = lt::bdecode(data, ec);
+#endif
+    }
+#endif
+
+#ifdef Q_OS_WIN
+    bool loadTorrentResumeData(QByteArray &data, CreateTorrentParams &torrentParams, int &queuePos, MagnetUri &magnetUri)
+#else
     bool loadTorrentResumeData(const QByteArray &data, CreateTorrentParams &torrentParams, int &queuePos, MagnetUri &magnetUri)
+#endif
     {
         lt::error_code ec;
 #if (LIBTORRENT_VERSION_NUM < 10200)
         lt::bdecode_node root;
         lt::bdecode(data.constData(), (data.constData() + data.size()), root, ec);
+#elif defined(Q_OS_WIN)
+        lt::bdecode_node root = lt::bdecode(data, ec);
 #else
         const lt::bdecode_node root = lt::bdecode(data, ec);
 #endif
         if (ec || (root.type() != lt::bdecode_node::dict_t)) return false;
 
         torrentParams = CreateTorrentParams();
 
+#ifdef Q_OS_WIN
+        fixBrokenSavePath(data, torrentParams, root);
```

Comment:
Why do you want to parse the data one more time in fixBrokenSavePath?

Reply:
Isn't the body of `fixBrokenSavePath()` self explanatory? In short we need to correct the path before we pass it libtorrent and set a flag to save it.
The data need to be parsed again because `bdecode_node` isn't mutable. It is just a view. So we judggle between bdecode_node->entry->bdecode_node.

PS: We could only do `bdecode_node->entry` and then use entry for parsing but that would require even more code and more branches.